dkg/frostp2p.go

```diff
@@ -32,13 +32,12 @@ var (
 )
 
 // newFrostP2P returns a p2p frost transport implementation.
-func newFrostP2P(tcpNode host.Host, peers map[peer.ID]cluster.NodeIdx, secret *k1.PrivateKey, threshold int) *frostP2P {
+func newFrostP2P(tcpNode host.Host, peers map[peer.ID]cluster.NodeIdx, secret *k1.PrivateKey, threshold, numVals int) *frostP2P {
 	var (
 		round1CastsRecv = make(chan *pb.FrostRound1Casts, len(peers))
 		round1P2PRecv   = make(chan *pb.FrostRound1P2P, len(peers))
 		round2CastsRecv = make(chan *pb.FrostRound2Casts, len(peers))
 
-		mu               sync.Mutex
```

Comment:
`bcastCallback` and `p2pCallback` now use two different mutexes instead of sharing `mu`.

Is this expected? Will this bite us in the back down the road?

Reply:
yes true, there are two different mutexes now. I think this is fine and AFAICT doesn't introduce a bug.

The mutex for `bcastCallback` guards the dedup maps for broadcast:
```go
dedupRound1Casts = make(map[peer.ID]bool)
dedupRound2Casts = make(map[peer.ID]bool)
```
While the mutex for `p2pCallback` guards the dedup map for P2P:
```go
dedupRound1P2P   = make(map[peer.ID]bool)
```

Reply:
But, this makes me wonder if we should initialize these maps inside the callback itself instead of passing down as parameters..

Reply:
okay, I'm pushing a change to share mutex instead of declaring new mutex instances inside each function. Since it's better for one mutex to guard all 3 maps:
```go
mu               sync.Mutex
dedupRound1Casts = make(map[peer.ID]bool)
dedupRound1P2P   = make(map[peer.ID]bool)
dedupRound2Casts = make(map[peer.ID]bool)
```

Reply:
If anything it's better not to alter the execution flow if it worked well up until now, although I believe that giving each closure their own maps + mutexes is still fine.